src/extension/wmts-multi-dimensional/src/main/java/org/geoserver/gwc/wmts/dimensions/Dimension.java

```diff
@@ -148,6 +216,14 @@ public Tuple<String, List<Integer>> getHistogram(Filter filter, String resolutio
 
             Tuple<String, List<Range>> specAndBuckets =
                     HistogramUtils.getNumericBuckets(min, max, resolutionSpec);
+
+            if (hasEndAttribute())
+                return getRangeHistogram(
+                        specAndBuckets,
+                        dimensionAttributeName,
+                        dimensionInfo.getEndAttribute(),
+                        query);
+
```

Comment:
I see the exact same code in the branch handlng dates.. if that's the case, best to check if we are dealing with ranges before entiring the number/date if, and just call the range histogram building once, rather than duplicating code?

Reply:
fixed